presets/cisco/firepower/preset.yaml

```diff
@@ -0,0 +1,94 @@
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 2)
+        - name: src_ip
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 3)
+        - name: src_port
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 4)
+        - name: dst_if
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 6)
+        - name: dst_ip
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 7)
+        - name: dst_port
+          expr: regexp_extract(message_text, r'((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)\\s+to\\s+((\\S+):)?(\\d+.\\d+.\\d+.\\d+)/(\\d+)', 8)
+        - name: ocsf_activity_id
+          expr: CASE WHEN message_number IN ('725001') THEN 1 WHEN message_number IN ('725007') THEN 2 ELSE 6 END
+        - name: ocsf_severity_id
+          expr: CASE WHEN level IN ('6', '7') THEN 1 WHEN level IN ('3', '4', '5') THEN 3 WHEN level IN ('0', '1', '2') THEN 4 END
+        - name: ocsf_severity
+          expr: CASE WHEN level IN ('6', '7') THEN 'Informational' WHEN level IN ('3', '4', '5') THEN 'Medium' WHEN level IN ('0', '1', '2') THEN 'High' END
+        - name: ocsf_src_endpoint_port
+          expr: CASE WHEN src_port = '' THEN NULL ELSE CAST(src_port AS INT) END
+        - name: ocsf_dst_endpoint_port
+          expr: CASE WHEN dst_port = '' THEN NULL ELSE CAST(dst_port AS INT) END
+      utils:
+        unreferencedColumns:
+          preserve: true
+          embedColumn: unmapped
+          embedColumnType: struct
+gold:
+  - name: network_activity
+    input: cisco_firepower
+    fields:
+      # OCSF required fields
+      - name: activity_id
+        from: ocsf_activity_id
+      - name: category_uid
+        expr: CAST('4' AS INT)
+      - name: class_uid
+        expr: CAST('4001' AS INT)
+      - name: time
+        from: time
+      - name: metadata.version
+        literal: 1.4.0
+      - name: severity_id
+        from: ocsf_severity_id
+      - name: type_uid
+        expr: CAST(class_uid * 100 + activity_id AS BIGINT)
+      # Other fields
+      - name: src_endpoint.interface_name
+        from: src_if
+      - name: src_endpoint.ip
+        from: src_ip
+      - name: src_endpoint.port
+        from: ocsf_src_endpoint_port
+      - name: dst_endpoint.interface_name
+        from: dst_if
+      - name: dst_endpoint.ip
+        from: dst_ip
+      - name: dst_endpoint.port
+        from: ocsf_dst_endpoint_port
+      - name: activity_name
+        literal: authentication
+      - name: app_name
+        literal: Cisco
+      - name: message
+        from: message_text
+      - name: severity
+        from: ocsf_severity
```

Comment:
Hmm... Both presets are writing from `cisco_firepower` into the `network_activity`.  If both datasources are created we will have two streams each with different checkpoints.  Wont this create duplicate entries in the gold table?

Reply:
Allen still need his coffee